iati_datastore/iatilib/parse.py

```diff
@@ -401,40 +401,23 @@ def default_language(xml, resource=None, major_version='1'):
     return codelists.by_major_version[major_version].Language.from_string(xml_value)
 
 
-def _open_resource(xml_resource, detect_encoding=False):
+def _open_resource_from_file(filename):
     """
-    Must return bytes object.
+    Expects a filename.
+    Returns a bytes file object.
     """
-    if isinstance(xml_resource, str):
-        # If it is a string...
-        if detect_encoding:
-            encoding = chardet.detect(xml_resource)['encoding']
-            if encoding in ('UTF-16LE', 'UTF-16BE'):
-                xml_resource = xml_resource.decode('UTF-16').encode('utf-8')
-
-        try:  # https://github.com/IATI/iati-datastore/issues/160
-            xml_resource_is_path = os.path.exists(xml_resource)
-        except TypeError:
-            xml_resource_is_path = False
-
-        # Previously included a workaround to this bug, but now appears to be fixed
-        # https://bugzilla.redhat.com/show_bug.cgi?id=874546
-        # appears now to be fixed
-        if xml_resource_is_path:
-            xmlfile = open(xml_resource, 'rb')
-        else:
-            xmlfile = BytesIO(xml_resource.encode())
+    if os.path.exists(filename):
```

Comment:
tbh I think you could probably skip the if…else…exception and just return open. It’ll raise a `FileNotFoundError` anyway if there’s a problem.